pkg/controller/actuator.go

```diff
@@ -261,10 +265,27 @@ func (a *actuator) Delete(ctx context.Context, log logr.Logger, ex *extensionsv1
 	namespace := ex.GetNamespace()
 	log.Info("Component is being deleted", "component", "", "namespace", namespace)
 
+	var istioNamespace string
+
 	istioNamespace, _, err := a.findIstioNamespaceForExtension(ctx, ex)
-	if err != nil {
+	if client.IgnoreNotFound(err) != nil {
 		return err
 	}
+	if apierrors.IsNotFound(err) {
+		exState, err := getExtensionState(ex)
+		if err != nil {
+			return err
+		}
+		if exState.IstioNamespace == nil {
+			// we have never reconciled this cluster completely, therefore no
+			// cleanup needs to be performed
+			return nil
```

Comment:
We should run `deleteSeedResources(ctx, log, namespace)` also if there is no istioNamespace set. Maybe we just do it at first while the deletion.

Reply:
restructured the order within the delete function. PTAL